docs/developers/strategies/architecture.md

```diff
@@ -159,3 +159,85 @@ Below are some of the user functions or properties under `OrderTracker` that you
     Returns a dictionary of order IDs that are being cancelled.
 
     Return type: `Dict[str, float]`
+
+
+## Fee Accounting
+
+### OrderCandidate
+
+An order proposal created by a strategy.
+
+A `BudgetChecker` takes an `OrderCandidate` and fills in fees to be paid for this particular order (in pre-defined tokens). Then checks if, after accounting for the fees, the account balances allow for a placement of this order, and if not, adjusts the order amount accordingly. 
+
+Fees can be payable in the base tokens, quote tokens, or 3rd party tokens.
+
+If an order opens a position, fees will be charged as an additional cost of the order.
+
+If an order closes a position, fees will be deducted from the returns.
```

Comment:
This part is generally, but not always, the case. For instance, Binance deducts the buy-order fees from the returns of a trade (e.g. if I create an order to buy 1,000 USDT worth of BTC, I get charged 1,000 USDT, but I will only get 999 USDT worth of BTC, with 1 USDT being the fee).

The user doesn't need to worry about this though, as each exchange can define their own fees schema and specify this behaviour. That being said, the docs should mention that the strategy has to get the budget checker from the exchange (i.e. `self.exchange.budget_checker`)